### ISSUE TITLE
include sorting sometimes adds undesirable blank lines

### ISSUE THREAD

**Issue by justusranvier** [include-what-you-use/include-what-you-use#1237]
Occasionally I encounter a situation where I need to add compiler pragmas around `#include` lines to suppress false positive warnings on code I can't modify.

Every time this happens IWYU will add an extra blank line when it sorts the include that I can't find a way to suppress.

Before:

```
#include <boost/multiprecision/cpp_int.hpp>
#include <algorithm>
#include <cctype>
#include <exception>
#include <limits>
#include <locale>
#include <sstream>
#include <utility>
```

Desired result:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Warray-bounds="
#include <boost/multiprecision/cpp_int.hpp>
#pragma GCC diagnostic pop
#include <algorithm>
#include <cctype>
#include <exception>
#include <limits>
#include <locale>
#include <sstream>
#include <utility>
```

Actual result:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Warray-bounds="
#include <boost/multiprecision/cpp_int.hpp>

#pragma GCC diagnostic pop
#include <algorithm>
#include <cctype>
#include <exception>
#include <limits>
#include <locale>
#include <sstream>
#include <utility>
```

If there is some deep architectural reason why it simply must add a blank line then at least if the line went *after* the `#pragma GCC diagnostic pop` it wouldn't be so bad.

**Comment by kimgr** [comment]
Yep, it looks a little dirty. Did you only run `include-what-you-use`, or did you use `fix_includes.py` to apply changes?

**Comment by justusranvier** [comment]
that is after running fix_includes.py

**Comment by kimgr** [comment]
Finally found some time to hack on this. Please take a look at #1374 and let me know what you think.

**Comment by justusranvier** [comment]
Thanks. I tried to build that commit so I could see how works on my project but it seems like I'll need a newer version of llvm before I can compile it so it may be a while before I can set that up.

**Comment by kimgr** [comment]
@justusranvier the fix is all python, so you should be able to try fix_includes.py from the pr branch directly

**Comment by justusranvier** [comment]
It does eliminate the blank line above the ```#pragma GCC disagnostic pop``` in my tests.

**Comment by kimgr** [comment]
Fixed on master as of [469b96f](https://github.com/include-what-you-use/include-what-you-use/commit/469b96f8a615d148d24605e63164d40c2c8b3260)
